core/src/cf/writer.rs

```diff
@@ -75,26 +75,36 @@ impl Writer {
 		current: Cow<'_, Value>,
 		store_difference: bool,
 	) {
+		warn!("Record cf called: {current:?}");
 		if current.is_some() {
 			self.buf.push(
 				ns.to_string(),
 				db.to_string(),
 				tb.to_string(),
 				match store_difference {
 					true => {
-						let patches = current.diff(&previous, Idiom(Vec::new()));
-						let new_record = !previous.is_some();
-						trace!("The record is new_record={new_record} because previous is {previous:?}");
 						if previous.is_none() {
 							TableMutation::Set(id, current.into_owned())
 						} else {
-							TableMutation::SetWithDiff(id, current.into_owned(), patches)
+							// We intentionally record the patches in reverse (current -> previous)
+							// because we cannot otherwise resolve operations such as "replace" and "remove".
+							let patches_to_create_previous =
+								current.diff(&previous, Idiom::default());
+							TableMutation::SetWithDiff(
+								id,
+								current.into_owned(),
+								patches_to_create_previous,
+							)
 						}
 					}
-					false => TableMutation::Set(id, current.into_owned()),
+					false => {
+						warn!("Store difference is off");
```

Comment:
If the warn is logged and may used by the end user, shall we printout the id of the record at least?

Reply:
Ah there should be no warns 😂 Debug info, thanks!